### ISSUE TITLE
aria-hidden="true" remains set on view samples dialog

### ISSUE THREAD

**Issue by joanmarie** [mathjax/MathJax-website#40]
Hi! I've been working on implementing support for MathML in the Orca screen reader and was going to use your samples dialog to demonstrate the current state. But Orca won't present your dialog or its contents. :( It appears that `aria-hidden="true"` remains set on the dialog which appears when you activate the "view samples" link. See attached screenshot.

As stated in the [ARIA spec](https://w3c.github.io/aria/aria/aria.html#aria-hidden):

> An element is considered hidden if it, or any of its ancestors are not rendered or have their aria-hidden attribute value set to true.

It would be great if `aria-hidden="true"` were only set on this element when it is not being shown to users.

Thanks in advance!

![mathjax-aria-hidden](https://cloud.githubusercontent.com/assets/3228171/10471798/78fd6e44-71e7-11e5-917c-91acbc7829ac.png)

**Comment by ZoeBijl** [comment]
Modals have `display: none` set by default, so is `aria-hidden` really necessary?

![Screenshot of web inspector shows .modal and .popover both have display set to none by default](https://cloud.githubusercontent.com/assets/5457769/10484163/cdd88f42-7283-11e5-89ec-bc14fe30a69d.png)

**Comment by StommePoes** [comment]
@MichielBijl they might have not wanted to worry about whatever those edge cases are where aria-labelledby still announces stuff even if it's display: none (like hidden-with-display-none error messages in a form hooked up to inputs via labelledby).

So nah, whatever JS function sets the display to block should just have another line to set the aria-hidden to false, should be good then.

**Comment by ZoeBijl** [comment]
@StommePoes good point.

**Comment by pkra** [comment]
Hi @Joanmarie thanks for raising the issue. Not sure if that's our fault or an upstream problem in Bootstrap's modal. I'm traveling this week but will look into it as soon as I can.

It might be better anyway to use the cleaner examples from our repository which are also available at http://cdn.mathjax.org/mathjax/latest/test/examples.html.

I suspect you might be interested in the upcoming release, currently in public beta, i.e.,  see http://beta.mathjax.org/mathjax/latest/test/examples.html for the samples with the beta release. That release introduces an accessibility extension that exposes the MathML source in the DOM so AT can pick it up (MathJax output isn't MathML after all); it also offers an accessible menu. It's not ideal (disconnecting visual and AT) but it's better than nothing until we can provide a full on-the-fly accessibility solution (which we're also working on).

We'd be very interested in any feedback on the beta and would be glad to add orca results to our documentation at http://docs.mathjax.org/en/v2.6-latest/misc/accessibility-features.html.

**Comment by joanmarie** [comment]
@pkra: Thanks for the info! I'll check it out and follow up with you.

Regarding the issue I'm reporting, I don't know a thing about design and associated tools, but based on your tip, I googled. My guess is that this may be due to a not-quite-right interpretation of the Bootstrap docs.

I downloaded their code and grepped for "aria-hidden". It's in their docs and tests, but not in their code. There are other ARIA properties used in their code (e.g. "aria-expanded" for dropdowns) so Bootstrap does provide some accessibility support.

As for their [modal docs](http://getbootstrap.com/javascript/#modals): The examples ("Static example" and "Live demo") are each accessible. They do use `aria-hidden=true` -- but not on the modal itself; instead, it is set on the text span serving as the Close button. That keeps the &times; char out of accessibility trees (the parent button element is what gets exposed to ATs and presented to users). HOWEVER, just below that, there's a section called "Make modals accessible" which states:

> Be sure to add `role="dialog"` and `aria-labelledby="..."`, referencing the modal title, to `.modal`, and `role="document"` to the `.modal-dialog` itself.

I don't think it's unreasonable to conclude from the above documentation that "ARIA stuff" belongs on the modal itself. And even `aria-hidden="true"` would make sense on the modal (though as @MichielBijl states, it seems redundant) _as long as it gets removed/set to `false` when the modal is shown to users_.

Long way of saying, I believe this is your bug; not Bootstrap's. Though Bootstrap perhaps should add something regarding this issue to their docs. Because if your team did it, my guess is that others have as well.
